aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelTree.java

```diff
@@ -0,0 +1,157 @@
+        }
+        Map<ModelSchema, Set<String>> descendants = new LinkedHashMap<>();
+        ModelSchema rootSchema = registry.getModelSchemaForModelInstance(root.iterator().next());
+        Set<String> rootIds = new HashSet<>();
+        for (T model : root) {
+            rootIds.add(model.getId());
+        }
+        recurseTree(descendants, rootSchema, rootIds);
+        return descendants;
+    }
+
+    private void recurseTree(
+            Map<ModelSchema, Set<String>> tree,
+            ModelSchema modelSchema,
+            Collection<String> parentIds
+    ) {
+        SQLiteTable parentTable = SQLiteTable.fromSchema(modelSchema);
+        for (ModelAssociation association : modelSchema.getAssociations().values()) {
+            switch (association.getName()) {
+                case "HasOne":
+                case "HasMany":
+                    String childModel = association.getAssociatedType(); // model name
+                    ModelSchema childSchema = registry.getModelSchemaForModelClass(childModel);
+                    SQLiteTable childTable = SQLiteTable.fromSchema(childSchema);
+                    String childPrimaryKey = childTable.getPrimaryKey().getAliasedName();
+                    QueryField queryField = QueryField.field(parentTable.getPrimaryKeyColumnName());
+
+                    // Chain predicates with OR operator.
+                    // No predicate = Match NONE.
+                    // 1 predicate = Match SELF.
+                    // 2 or more predicates = Match ANY.
+                    QueryPredicate predicate = QueryPredicates.none();
+                    for (String parentId : parentIds) {
+                        QueryPredicateOperation<Object> operation = queryField.eq(parentId);
+                        if (QueryPredicates.none().equals(predicate)) {
+                            predicate = operation;
+                        } else {
+                            predicate = operation.or(predicate);
+                        }
+                    }
+
+                    // Collect every children one level deeper than current level
+                    // SELECT * FROM <CHILD_TABLE> WHERE <PARENT> = <ID_1> OR <PARENT> = <ID_2> OR ...
+                    QueryOptions options = Where.matches(predicate);
+                    Set<String> childrenIds = new HashSet<>();
+                    try (Cursor cursor = queryAll(childModel, options)) {
+                        if (cursor != null && cursor.moveToFirst()) {
+                            int index = cursor.getColumnIndexOrThrow(childPrimaryKey);
+                            do {
+                                childrenIds.add(cursor.getString(index));
+                            } while (cursor.moveToNext());
+                        }
+                    } catch (DataStoreException exception) {
+                        // Don't cut the search short. Populate rest of the tree.
```

Comment:
Looks like this will only be thrown if `SQLPredicate::parsePredicate` encounters a `QueryPredicate` that it can't parse, which should only happen if there is a bug in the library.  Wouldn't it be better to throw an exception back to the user, so they can report the bug to us, instead of failing silently?

Reply:
it can also happen from failure to query the SQLite database, which can result from various situations (cursor size overflow, too many cursors, corrupt row in cursor, etc.).

Reply:
Hmm.  I still think a customer would probably want to know about all of those failures, so they can report them to us, so we can fix those issues.  How about adding a `LOG.error` here?

Reply:
Sure thing. It's never bad to provide some more insight i guess